🔒 Increase PBKDF2 iterations to enhance password hashing security

AlmosGH opened the pull request:
This pull request includes a crucial update to the password generation function to enhance security by increasing the number of iterations in the PBKDF2 key derivation process.

Security enhancement:

* [`popup/popup.js`](diffhunk://#diff-2ce48ac6dbf8501b60a7bbeb4c05cbd8b5eead20f4566669b124ade52182387bL151-R151): Increased the number of iterations in the PBKDF2 key derivation function from 300,000 to 1,000,000 to strengthen password security.

AlmosGH committed:
🔒 Increase PBKDF2 iterations to enhance password hashing security

Xeimyn commented:
Would be better to add a setting instead. 

1.000.000 iterations also create a performance issue. Since the password is updated on every input change it lags behind up to 2 seconds. Probably have to add some "debounce" or whatever

AlmosGH committed:
Add a setting for PBKDF2 iterations

Add a setting for PBKDF2 iterations to address performance issues.

* Add a new input field for PBKDF2 iterations in `settings/settings.html`.
* Add event listeners for the new input field in `settings/settings.js`.
* Update the `genPW` function in `popup/popup.js` to use the PBKDF2 iterations setting from localStorage.
* Add default PBKDF2 iterations setting in `settings/settings.js`.
* Add styles for the new PBKDF2 iterations input field in `settings/settings.css` and `popup/popup.css`.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/AlmosYT/LesserPass?shareId=XXXX-XXXX-XXXX-XXXX).

AlmosGH committed:
Add warning message for PBKDF2 iterations below 300000

* **settings/settings.html**
  - Add warning message element for PBKDF2 iterations below 300000.

* **settings/settings.js**
  - Display warning message if PBKDF2 iterations are below 300000.

* **popup/popup.js**
  - Add debounce function to `regeneratePassword` function.
  - Display alert if PBKDF2 iterations are below 300000 in `genPW` function.

AlmosGH committed:
Merge pull request #1 from AlmosYT/add-pbkdf2-iterations-setting

Add a setting for PBKDF2 iterations

AlmosGH commented:
@Xeimyn done

Xeimyn commented:
```js
minusButtonIterations.addEventListener("click",function() {
		if (iterationsElement.value >= 50000) {
			iterationsElement.value -= 50000
		}
	}
```

Yeah man that makes total sense 👍
